exercises/simple-cipher/canonical-data.json

```diff
@@ -0,0 +1,123 @@
+{
+  "exercise": "simple-cipher",
+  "version": "1.0.0",
+  "cases": [
+    {
+      "description": "Random key cipher",
+      "property": "random",
+      "cases": [
+        {
+          "description": "It has a key made of letters",
+          "property": "key",
+          "input": "any",
+          "matches_regex": "^[a-z]+$"
+        },
+        {
+          "description": "It can encode",
+          "property": "encode",
+          "input_plaintext": "aaaaaaaaaa",
+          "output_key": "cipher.key"
+        },
+        {
+          "description": "It can decode",
+          "property": "decode",
+          "input_key": "cipher.key",
+          "output_string": "aaaaaaaaaa"
+        },
+        {
+          "description": "It is reversible",
+          "property": "encode -> decode",
+          "input_encode": "plaintext string",
+          "output_decode": "plaintext string"
+        }
+      ]
+    },
+    {
+      "description": "Incorrect key cipher",
+      "property": "incorrect",
+      "cases": [
+        {
+          "description": "It throws an error with an all caps key",
```

Comment:
Maybe this test should be about testing a string with a single uppercase character, as that is the minimal counter-example.